db.ts

```diff
@@ -6,6 +6,7 @@ export const products: IProduct[] = [
     price: 10,
+    discountPrice: 34,
```

Comment:
como o desconto é percentual, acho que poderíamos chamar a propriedade apenas de `discount`

Reply:
Não sei se estou certo, mas acho que tentaria resolver dessa forma:

`export default interface IProduct {
  id: number;
  title: string;
  img: string;
  price: number;
  discount: number;
  discountedPrice?: number; // tentaria criar uma computed property
  description: string;
  fullDescription: string;
}

export class Product implements IProduct {
  public id: number;
  public title: string;
  public img: string;
  public price: number;
  public discount: number;
  public description: string;
  public fullDescription: string;

  constructor(product: IProduct) {
    this.id = product.id;
    this.title = product.title;
    this.img = product.img;
    this.price = product.price;
    this.discount = product.discount;
    this.description = product.description;
    this.fullDescription = product.fullDescription;
  }

  get discountedPrice() {
    return this.price - (this.price * this.discount) / 100;
  }
}`

Reply:
E como resultado o banco de dados ficaria desse jeito:

`import IProduct, { Product } from "./entities/IProduct";

export const products: IProduct[] = [
  new Product({
    id: 3,
    title: "Camiseta 3",
    img: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR6XGdJw5NsYRRklb8YpamARtn8YTG5aOXVhg&usqp=CAU",
    price: 10,
    discount: 10,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 4,
    title: "Camiseta 3",
    img: "https://aramismenswear.vtexassets.com/arquivos/ids/742946-800-auto?v=637525574779270000&width=800&height=auto&aspect=true",
    price: 10,
    discount: 20,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 5,
    title: "Camiseta 5",
    img: "https://d2cnvqfvkdm3ii.cloudfront.net/Custom/Content/Products/16/44/1644118_blusa-le-lis-blanc-devore-i-feminina-11-11-3731_t18_637469137371272089.jpg",
    price: 10,
    discount: 30,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
  new Product({
    id: 15,
    title: "Caneta 17",
    img: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTsxzknPvCjyu0l2mlfxKhmv3xy3YzSPSukLg&usqp=CAU",
    price: 10,
    discount: 0,
    description:
      "It is a long established fact that a reader will be distracted.",
    fullDescription:
      "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using Content here, content here, making it look like readable English. ",
  }),
];
`

Reply:
E acho que mudaria 'discountedPrice' para 'computedPrice'